File: src/main/java/com/paragon/api/util/calculations/Timer.java
package com.paragon.api.util.calculations;

public class Timer {

    public long milliseconds;

    public Timer() {
        this.milliseconds = -1;
    }

    public boolean hasMSPassed(double time) {
        return System.currentTimeMillis() - milliseconds > time;
    }

    public void reset() {
        this.milliseconds = System.currentTimeMillis();
    }

}

File: src/main/java/com/paragon/api/util/calculations/Timer.kt
package com.paragon.api.util.calculations

/**
 * @author Surge
 */
class Timer {

    private var milliseconds = -1L

    fun hasMSPassed(time: Double) = System.currentTimeMillis() - milliseconds > time

    fun reset() {
        milliseconds = System.currentTimeMillis()
    }

}